superset/tags/commands/create.py

```diff
@@ -60,3 +62,45 @@ def validate(self) -> None:
             )
         if exceptions:
             raise TagInvalidError(exceptions=exceptions)
+
+
+class CreateCustomTagWithRelationshipsCommand(CreateMixin, BaseCommand):
+    def __init__(self, data: dict[str, Any]):
+        self._tag = data["name"]
+        self._objects_to_tag = data.get("objects_to_tag")
+        self._description = data.get("description")
+
+    def run(self) -> None:
+        self.validate()
+        try:
+            tag = TagDAO.get_by_name(self._tag.strip(), TagTypes.custom)
+            if self._objects_to_tag:
+                TagDAO.create_tag_relationship(
+                    objects_to_tag=self._objects_to_tag, tag=tag
+                )
+
+            if self._description:
+                tag.description = self._description
+                db.session.commit()
+
+        except DAOCreateFailedError as ex:
+            logger.exception(ex.exception)
+            raise TagCreateFailedError() from ex
+
+    def validate(self) -> None:
+        exceptions = []
+        # Validate object_id
+        if self._objects_to_tag:
+            if any(obj_id == 0 for obj_type, obj_id in self._objects_to_tag):
+                exceptions.append(TagCreateFailedError())
```

Comment:
@hughhhh let's make this a different 422 error, not the same one that we're using for the 500.